wp-content/themes/pub/wporg-learn-2024/patterns/sensei-lesson-facilitator-notes.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * Title: Sensei Lesson Facilitator Notes
+ * Slug: wporg-learn-2024/sensei-lesson-facilitator-notes
+ * Inserter: no
+ */
+
+?>
+
+<input type="checkbox" id="wporg-learn-facilitator-notes-toggle">
```

Comment:
Need to consider the accessibility of this control and the modal content. At present this input isn't focusable with keyboard. We may want to move focus to the close button after activation. I'd consult some [modal best practice docs](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/).

On mobile it is definitely following a modal pattern, where focus should be trapped within the modal, and background page elements should not be interactive. On desktop it's arguable that the rest of the UI should remain interactive, as it's more of a panel.

@ryelle any thoughts on this?

Reply:
Also I expected this pattern would use a button element, hooked up to the interactivity API to control the modal visibility, however I'm not sure whether this input based approach is any less accessible. Might need more aria attributes though as I think it's a less typical pattern.

Reply:
> I expected this pattern would use a button element,

I think a button (with `aria-expanded` and `aria-controls` attributes) would be better. Right now this would read as checking a checkbox, with no way to know that it would reveal more content.

> On mobile it is definitely following a modal pattern, where focus should be trapped within the modal, and background page elements should not be interactive. On desktop it's arguable that the rest of the UI should remain interactive, as it's more of a panel.

If it's easier, you can make it a modal (with focus trapping) in both cases - and if someone clicks outside the panel to use the rest of the site, close the panel. Otherwise, on desktop, I would expect it to close when I tab through everything, and focus moves on to the next tab stop (from the screenshots, I assume that's "exit course"). The desktop global nav submenus work like this— if you open "Community" & start tabbing, the submenu closes when  "About" is focused.